src/misc_tools.h

```diff
@@ -146,6 +146,12 @@ void bytes_convert_str(char *buf, int size, uint64_t bytes);
 /* checks if a file exists. Returns true or false */
 bool file_exists(const char *path);
 
+/* Returns 0 if path points to a directory.
```

Comment:
This type of ad-hoc enum is a bit sad. You're erasing the information from named "S_IFREG" and turning it into integers with documented but unnamed meaning. Perhaps add an enum for this?

Reply:
I added an enum with 3 values, which can be expanded if ever necessary.